lib/models/debugs_bunny/trace.rb

```diff
@@ -1,18 +1,34 @@
 # frozen_string_literal: true
 
+require_relative 'application_record'
 require_relative 'concerns/can_generate'
+require_relative 'concerns/has_encrypted_attributes'
+require 'debugs_bunny/config'
+require 'debugs_bunny/internal/schema/column_descriptor'
 
 module DebugsBunny
   class Trace < DebugsBunny::ApplicationRecord
     include CanGenerate
+    include HasEncryptedAttributes
 
     self.abstract_class = true
 
     has_guid 'trc'
 
+    def generate_partition_guid
+      self.partition_guid = DebugsBunny.configuration.encryption_partition_guid
+    end
+
+    define_encrypted_attributes do
+      attr_encrypted :dump
+    end
+
     define_table do |t|
       t.define_column :guid, :string, null: false
-      t.define_column :dump, :string, null: false
+      t.define_column :encrypted_dump, :string
```

Comment:
Notice that the Trace table no longer has a field called `dump`; this has been replaced with the equivalent `encrypted_dump` field. A trace's dump attribute is still accessible through the same `.dump` interface.